src/test/java/com/tabnine/binary/BootstrapperSupportTests.java

```diff
@@ -0,0 +1,80 @@
+package com.tabnine.binary;
+import com.tabnine.binary.fetch.*;
+import com.tabnine.general.StaticConfig;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import java.util.List;
+import java.util.Optional;
+import java.util.prefs.BackingStoreException;
+import java.util.prefs.Preferences;
+import static com.tabnine.general.StaticConfig.versionFullPath;
+import static com.tabnine.testutils.TestData.PREFERRED_VERSION;
+import static com.tabnine.testutils.TestData.aVersions;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class BootstrapperSupportTests {
+    @Mock
+    private LocalBinaryVersions localBinaryVersions;
+    @Mock
+    private BinaryRemoteSource binaryRemoteSource;
+    @Mock
+    private BundleDownloader bundleDownloader;
+
+    @InjectMocks
+    private BinaryVersionFetcher binaryVersionFetcher;
+
+    @BeforeEach
+    public void setUp() throws BackingStoreException {
+        Preferences preferences = Preferences.userNodeForPackage(BootstrapperSupport.class);
```

Comment:
see my comment  regarding the Preferences.userNodeForPackage impl coupling

Reply:
where are the comments?

Reply:
https://github.com/codota/tabnine-intellij/pull/68#discussion_r517401362